sampo/scheduler/genetic/schedule_builder.py

```diff
@@ -189,7 +186,7 @@ def build_schedule(wg: WorkGraph,
         # save best individuals
         hof = tools.HallOfFame(1, similar=compare_individuals)
 
-        fitness_f = fitness_constructor(native.evaluate)
+        fitness_f = fitness_constructor(native.evaluate) if not have_deadline else TimeFitness(native.evaluate)
```

Comment:
`fitness_constructor` is already parametrization about fitness function.
You shouldn't have any overrides of it. 
If you are going to do something like this, do it in `GeneticScheduler` class.

Reply:
if a deadline is given, then a two-stage lexicographic genetic algorithm is applied. Therefore, at the first stage, optimization is carried out over Time, and already at the second stage, the function passed to fitness_constructor is optimized.